### PR TITLE
Show OCR options to non-premium users

### PR DESCRIPTION
**Pull request by eyeseast**
Screenshot 2023-03-29 at 3 29 04 PM
Screenshot 2023-03-29 at 3 29 14 PM

### CODE REVIEW

File: src/pages/app/Paginator.svelte [hunk 1]
```diff
@@ -1,16 +1,10 @@
 <script>
-  import { search, initSearch, searchNext, searchPrev } from "@/search/search";
-  import { pushUrl } from "@/router/router";
-  import { queryBuilder } from "@/util/url";
-  import { simplePlural } from "@/util/string";
```
Comment (jimkang) [inline src/pages/app/Paginator.svelte:5]: Always great to see dead code removed!

**Review by jimkang — CHANGES_REQUESTED** [review]
Other than the concerns I had about nothing showing up in the language control in Firefox (which is caused by Firefox's datalist implementation as you said), I think this mostly looks good.

There are some simplification changes that I think would help with maintainability, but it's possible I'm actually misunderstanding the code. If that's the case just let me know!

File: src/common/Select.svelte [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  export let name;
+  export let label;
+  export let placeholder = "";
+  export let options = []; // [value, name, disabled]
+  export let selected;
+  export let value = null;
+
+  let input;
+
+  $: valueToName = options.reduce((m, opt) => {
```
[the remark this answers is not in the document]
Comment (eyeseast) [reply src/common/Select.svelte:11]: It's not. Thought I'd need it but haven't yet. Might just take it out.

File: src/common/Select.svelte [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    m.set(opt[0], opt[1]);
+    return m;
+  }, new Map());
+
+  $: nameToValue = options.reduce((m, opt) => {
+    m.set(opt[1], opt[0]);
+    return m;
+  }, new Map());
```
[the remark this answers is not in the document]
Comment (jimkang) [reply src/common/Select.svelte:19]: Ah, OK. Maybe a comment explaining that this Map is for finding out if an option is selected and only for that would help.

**Review by jimkang — APPROVED** [review]
Latest commit looks good!

**Pull request merged**